src/client/automation/playback/upload.js

```diff
@@ -1,12 +1,37 @@
-import { doUpload } from '../deps/hammerhead';
+import {
+    doUpload, eventSandbox, nativeMethods,
+} from '../deps/hammerhead';
 import { arrayUtils } from '../deps/testcafe-core';
 
 
+const REQUIRED_ATTR_NAME     = 'required';
+const FORM_SUBMIT_EVENT_NAME = 'submit';
+
 export default class UploadAutomation {
-    constructor (element, paths, createError) {
+    constructor (element, paths, createError, isNativeAutomation) {
         this.element     = element;
         this.paths       = paths;
         this.createError = createError;
+
+        if (isNativeAutomation)
+            this._handleRequiredInput();
+    }
+
+    _handleRequiredInput () {
+        const isRequired = nativeMethods.hasAttribute.call(this.element, REQUIRED_ATTR_NAME);
+
+        if (isRequired)
+            nativeMethods.removeAttribute.call(this.element, REQUIRED_ATTR_NAME);
```

Comment:
As I see below you restore this attribute only if an input is in a form. So I have 2 questions: 
1. Are you sure that it isn't necessary to restore this attribute for non-form inputs? 
2. Are you sure that we really need to remove this attribute for non-form inputs?

Based on the answers we should skip calling this method for non-form inputs or create a separate test for this case.

Reply:
@Aleksey28 Could you please check these cases? I think it should not be a problem.

Reply:
We can add an input with a file on a page but as I see in the issue it is needed to remove this attribute only during submitting a form. Let's check `this.element.form` before calling this method and remove the checking a form below in this method.

Reply:
Ok, @Bayheck please add additional condition before the `this._handleRequiredInput()` call. Let's ensure that the `required` attribute is removed only in forms. After that, please check this approach with user's usage scenario.

Reply:
I updated the code